### PR TITLE
Reuse cached dev auth context in CLI

### LINKED ISSUE #998: [Bug]: CLI passkey + 2FA login fails with an opaque 401 when local dev context is not exported
## Issue Type
bug

## Phase
implementation

## Version
main (local checkout observed on 2026-03-29)

## Description
Following the canonical passkey + 2FA CLI login steps from `docs/guide/local-dev-auth-login.md` does not work from a fresh shell.

After `mise run dev` prepares the local auth context, running:

```bash
cargo run -q -p ugoite-cli -- config set --mode backend --backend-url http://localhost:8000
cargo run -q -p ugoite-cli -- auth login --username dev-local-user --totp-code <current code>
```

returns:

```text
Error: HTTP 401 Unauthorized: {"detail":"Passkey-bound local context is missing or invalid."}
```

The root cause is that `ugoite-cli/src/http.rs` only forwards `x-ugoite-dev-passkey-context` when `UGOITE_DEV_PASSKEY_CONTEXT` is already exported, but the CLI login section in `docs/guide/local-dev-auth-login.md` does not tell users to source the helper exports in the shell where they run `ugoite auth login`.

## Steps to Reproduce
1. Start local development with `mise run dev` and complete the terminal username + 2FA prompt.
2. Open a second shell that has **not** sourced the local dev auth helper exports.
3. Run `cargo run -q -p ugoite-cli -- config set --mode backend --backend-url http://localhost:8000`.
4. Run `cargo run -q -p ugoite-cli -- auth login --username dev-local-user --totp-code <current code>`.
5. Observe the 401 error about missing or invalid passkey-bound local context.

## Expected Behavior
A first-time user following the documented CLI login flow should either:

- succeed without extra hidden shell state, or
- receive an actionable error that explicitly says the CLI needs `UGOITE_DEV_PASSKEY_CONTEXT` (and, when applicable, `UGOITE_DEV_AUTH_PROXY_TOKEN`) plus an exact command to export the required values in the current shell.

## Environment
- Repository: `ugoite/ugoite`
- Branch/version: `main` local checkout on 2026-03-29
- OS: Linux devcontainer
- Auth mode: `passkey-totp`
- Backend/frontend started with `mise run dev`
- Related docs/code: `docs/guide/local-dev-auth-login.md` (CLI login section), `ugoite-cli/src/commands/auth.rs`, `ugoite-cli/src/http.rs`

### PR DESCRIPTION
**Pull request by tohboeh5**
## Summary
- reuse the cached local dev auth file for passkey-totp CLI login when UGOITE_DEV_PASSKEY_CONTEXT is not exported
- scope cached passkey-context fallback to loopback/local development endpoints only, and align the CLI help text and guide docs with the recovery flow
- update REQ-OPS-015 and HTTP helper coverage for cached-context fallback and blank-cache edge cases

## Related Issue (required)
closes #998

## Testing
- manual local backend plus fresh-shell CLI login using only UGOITE_DEV_AUTH_FILE
- CARGO_BUILD_JOBS=1 RUSTFLAGS='-C debuginfo=0' cargo test -p ugoite-cli --test test_cli_req_ops_006_http_helpers -- --nocapture
- CARGO_BUILD_JOBS=1 RUSTFLAGS='-C debuginfo=0' cargo test -p ugoite-cli --test test_auth test_cli_auth_login_req_ops_015_ -- --nocapture
- uv run --with pytest --with pyyaml --with bashlex pytest docs/tests/test_guides.py -W error -v -k test_docs_req_ops_015_local_dev_auth_docs_cover_manual_modes
- TMPDIR=/workspace/tmp TMP=/workspace/tmp TEMP=/workspace/tmp mise run test

- [x] I ran the applicable checks for these changes.

### CODE REVIEW

File: ugoite-cli/src/http.rs [hunk 1]
```diff
@@ -82,24 +90,52 @@ fn add_auth_headers(req: reqwest::RequestBuilder) -> reqwest::RequestBuilder {
...
     };
 
-    let Ok(context) = std::env::var("UGOITE_DEV_PASSKEY_CONTEXT") else {
-        return req;
-    };
-    let trimmed = context.trim();
-    if trimmed.is_empty() {
-        return req;
+    if let Some(context) =
+        non_empty_env_var("UGOITE_DEV_PASSKEY_CONTEXT").or_else(cached_dev_passkey_context)
```
Comment (tohboeh5) [inline ugoite-cli/src/http.rs]: `cached_dev_passkey_context()` is now wired into every `http_post_with_dev_auth_proxy()` call, including `auth login` against backend/api-mode endpoints that can be remote `https://...` URLs. That means a user who previously ran local dev can automatically leak the passkey-bound local context from `~/.ugoite/dev-auth.json` to any remote server they later point the CLI at. This fallback needs to be scoped to loopback/local-dev endpoints only (or gated behind an explicit opt-in), not applied unconditionally.
